src/Files.App/Actions/Navigation/OpenDirectoryInNewTabAction.cs

```diff
@@ -36,7 +38,7 @@ public OpenDirectoryInNewTabAction()
 
 		public async Task ExecuteAsync()
 		{
-			foreach (ListedItem listedItem in context.ShellPage.SlimContentPage.SelectedItems)
+			foreach (ListedItem listedItem in context.ShellPage?.SlimContentPage?.SelectedItems)
```

Comment:
I don't think this avoids NullReferenceException. If `context.ShellPage?.SlimContentPage` is null, we should return before `foreach`.

Reply:
Can someone call ExecuteAsync if IsExecutable is false?

Reply:
I don't think this code is reachable if SlimContentPage is null.

Reply:
Normally your thoughts are right, but the possibility that `SlimContentPage` is nulled by another thread at the moment `ExecuteAsync` is called should be considered.